### DISCUSSION TITLE
Adding Support for Firebird 4

### DISCUSSION THREAD

**Discussion by e-belair** [sqlalchemy/sqlalchemy#6742] [Usage Questions] [answered]
Hi, seems to miss support for Firebird 4, see the error message below.
Is there a temporary way to solve the issue?

```
self = <sqlalchemy.dialects.firebird.fdb.FBDialect_fdb object at 0x7fa2d3e508b0>
version = 'LI-T4.0.0.1963 Firebird 4.0 Beta 2'

    def _parse_version_info(self, version):
        m = match(
            r"\w+-V(\d+)\.(\d+)\.(\d+)\.(\d+)( \w+ (\d+)\.(\d+))?", version
        )
        if not m:
>           raise AssertionError(
                "Could not determine version from string '%s'" % version
            )
E           AssertionError: Could not determine version from string 'LI-T4.0.0.1963 Firebird 4.0 Beta 2'

env/lib/python3.8/site-packages/sqlalchemy/dialects/firebird/kinterbasdb.py:177: AssertionError
```
**Versions.**
 - OS: Ubuntu 20.04.2 LTS
 - Python: 3.8.10
 - SQLAlchemy: 1.4.20
 - Database: Firebird 4.0
 - DBAPI:

**Comment by CaselIT** [comment, marked as answer]
Hi,

As mentioned in the docs, the internal firebird dialect is deprecated.
Please use the external version at https://github.com/pauldex/sqlalchemy-firebird

**Reply by e-belair** [reply]
Ho ok, I'm sorry, I wasn't aware of an external package.
But I'm worried cause the package seems not to be very maintained and does not support FB4 too

**Reply by CaselIT** [reply]
Sadly we don't have the resources to maintain the firebird and sybase dialect. Even before the official deprecation they were best effort at best, and probably partially broken in various ways.

I'm sure the external library would welcome contributions.

Also cc @gordthompson

**Reply by e-belair** [reply]
I'll try to contribute if I can but I'm not an expert ^^"
